internal/mesh/internal/controllers/gatewayproxy/mapper/apigatewayworkloads.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package mapper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/mesh/internal/controllers/gatewayproxy/fetcher"
+	"github.com/hashicorp/consul/internal/resource"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v2beta1"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v2beta1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+var APIGatewaysInParentRefs = func(ctx context.Context, rt controller.Runtime, res *pbresource.Resource) ([]controller.Request, error) {
+	fetcher := fetcher.New(rt.Client)
+
+	requests := make([]controller.Request, 0)
+
+	route, err := fetcher.FetchTCPRoute(ctx, res.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	if route == nil {
+		return nil, nil
+	}
+
+	for _, parentRef := range route.Data.GetParentRefs() {
+		if !resource.EqualType(parentRef.Ref.Type, pbmesh.APIGatewayType) {
+			return nil, fmt.Errorf("parent reference type %s is not supported", parentRef.Ref.Type)
```

Comment:
Won't this actually bomb for legitimate TCPRoute parent references to services that we just don't care about?

Reply:
good call, I wrote this when we were still thinking that a route could only reference all gateways or all services as the parent refs